src/quo2/components/share/qr_code/view.cljs

```diff
@@ -1,12 +1,87 @@
 (ns quo2.components.share.qr-code.view
-  (:require [quo2.components.share.qr-code.style :as style]
+  (:require [quo2.components.avatars.account-avatar.view :as account-avatar]
+            [quo2.components.avatars.channel-avatar.view :as channel-avatar]
+            [quo2.components.avatars.user-avatar.view :as user-avatar]
+            [quo2.components.avatars.wallet-user-avatar :as wallet-avatar]
+            [quo2.components.share.qr-code.style :as style]
             [react-native.core :as rn]
-            [react-native.fast-image :as fast-image]))
-
-(defn qr-code
-  [{:keys [source width height]}]
-  [rn/view
-   {:style style/container}
-   [fast-image/fast-image
-    {:source source
-     :style  (style/image width height)}]])
+            [react-native.fast-image :as fast-image]
+            [utils.image-server :as image-server]))
+
+(defn- qr-code-image
+  [{:keys [size url media-server-port]}]
+  (let [media-server-uri (image-server/get-qr-image-uri-for-any-url
```

Comment:
I would prefer if we could keep the QR code component (and the component lib for that matter) less coupled with Status concepts, thus moving the responsibility to generate the image server URL to the caller of `qr-code/view`. We have a few options:

1. The caller passes a fully valid image server URL, ready to be used by `fast-image` (my preferred approach).
2. The caller passes an image component and the `qr-code/view` only worries about styling it inside a container.
3. The caller passes a thunk to `qr-code/view` that's responsible for returning a fully working image server URL.